x-pack/plugins/apm/public/components/shared/FilterBar/ServiceEnvFilter.tsx

```diff
@@ -0,0 +1,94 @@
+import { EuiFilterGroup, EuiFilterButton, EuiSelect } from '@elastic/eui';
+import React, { useEffect, useRef, useState } from 'react';
+import { useFetcher } from '../../../hooks/useFetcher';
+import { loadServiceEnvironments } from '../../../services/rest/apm/services';
+import { useLocation } from '../../../hooks/useLocation';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { history } from '../../../utils/history';
+import { fromQuery, toQuery } from '../Links/url_helpers';
+
+function without<T>(obj: T, key: string): T {
+  const result: { [objKey: string]: unknown } = Object.assign({}, obj);
+  delete result[key];
+  return result as T;
+}
+
+export const ServiceEnvFilter: React.FC = () => {
+  const location = useLocation();
+  const { urlParams } = useUrlParams();
+  const { start, end, kuery, environment } = urlParams;
+  const { data = [] } = useFetcher(
+    () => loadServiceEnvironments({ start, end, kuery, environment: '' }),
+    [start, end, kuery]
+  );
+  const availableEnvironments = ['Any', ...data, 'Not set'];
+  const [selectedValue, setSelectedValue] = useState(availableEnvironments[0]);
+  const isPostEffect = useRef(false);
+
+  function updateEnvironmentUrl(environment?: string) {
+    if (!environment || environment === availableEnvironments[0]) {
+      history.push({
+        ...location,
+        search: fromQuery(without(toQuery(location.search), 'environment'))
+      });
+    } else if (
+      environment === availableEnvironments[availableEnvironments.length - 1]
+    ) {
+      history.push({
+        ...location,
+        search: fromQuery({
+          ...toQuery(location.search),
+          environment: 'Null'
+        })
+      });
+    } else {
+      history.push({
+        ...location,
+        search: fromQuery({
+          ...toQuery(location.search),
+          environment
+        })
+      });
+    }
+  }
+
+  useEffect(
+    () => {
+      if (isPostEffect.current) {
+        if (environment !== selectedValue) {
+          updateEnvironmentUrl(selectedValue);
+        }
+      } else {
+        if (!environment && selectedValue !== availableEnvironments[0]) {
+          updateEnvironmentUrl(selectedValue);
+        } else if (
+          environment === 'Null' &&
+          selectedValue === availableEnvironments[0]
+        ) {
+          setSelectedValue(
+            availableEnvironments[availableEnvironments.length - 1]
+          );
+        } else if (environment && selectedValue === availableEnvironments[0]) {
+          setSelectedValue(environment);
+        }
+      }
+      isPostEffect.current = true;
+    },
+    [selectedValue, availableEnvironments.join(',')]
+  );
```

Comment:
This looks overly complicated. What are you trying to achieve?

Reply:
this component is just a placeholder for the final one...i used the basic EuiSelect while i was implementing the filters on the back end. This whole component is subject to change.

Reply:
I think `EuiSelect` is fine to use for now. If it's simpler to use, then let's do that. We can always enhance it later.